primeqa/mitqa/utils/create_table_retriever_training_data.py

```diff
@@ -0,0 +1,152 @@
+        A list of dictionaries
+    """
+    data = json.load(open("data/ottqa/all_plain_tables.json"))
+    return data  
+
+def create_collections():
+    """
+    The create_collections function creates a tsv file with the following columns:
+        - id: The unique identifier for this table.
+        - text: The linearized version of the table, where each row is on its own line.
+        - title: The title of the Wikipedia page containing this table.
+    
+    Args:
+    
+    Returns:
+        A tsv file with the id, text and title of each table
+    """
+    data = load_all_tables()
+    tsv_file = open("data/ottqa/linearized_tables.tsv", 'w', encoding='utf8', newline='')
+    tsv_writer = csv.writer(tsv_file, delimiter='\t',lineterminator='\n')
+    for k,v in data.items():
+        id = k
+        text = linearize_table(k,data)
+        title = v['title']
+        tsv_writer.writerow([id,text,title])
+        
+        
+    
+
+def gen_triples_from_dict(split):
+    """
+    The gen_triples_from_dict function generates a TSV file containing the question, positive table, and negative table for each triple in the triples_dict.json file.
+    The TSV file is named &quot;triples_&lt;split&gt;.tsv&quot; where &lt;split&gt; is either 'train', 'dev', or 'test'.
+    
+    
+    Args:
+        split: Specify which split of the data to generate triples for
+    
+    Returns:
+        A tsv file with a question, the positive table and the negative table
+    """
+    all_tables = load_all_tables()
+    triples_data = json.load(open("data/ottqa/triples_dict_"+split+".json"))
+    tsv_file = open("data/ottqa/triples_"+split+".tsv", 'w', encoding='utf8', newline='')
+    tsv_writer = csv.writer(tsv_file, delimiter='\t', lineterminator='\n')
+    for k,v in tqdm(triples_data.items()):
+        question = k
+        pos_table_id = v['pos_table']
+        neg_table_id = v['neg_table']
+        linearized_pos = linearize_table(pos_table_id,all_tables)
+        linearized_neg = linearize_table(neg_table_id,all_tables)
+        tsv_writer.writerow([question,linearized_pos,linearized_neg])
+    
+
+def gen_pos_neg_table_id_dict(split):
+    """Generates training data for training dpr
+    """
+    data = json.load(open("data/ottqa/released_data/"+split+".json"))
+    question_table_id_dict = {}
+    all_table_ids = list(load_all_tables().keys())
+    neg_tables= all_table_ids.copy()
+    not_found = []
+    for d in tqdm(data):
+        question = d['question']
+        pos_table_id = d['table_id']
+        if pos_table_id not in all_table_ids:
+            not_found.append(pos_table_id)
+            continue
+        else:
+            neg_tables.remove(pos_table_id)
+        neg_table_id = random.choice(neg_tables)
+        question_table_id_dict[question] = {"pos_table":pos_table_id,"neg_table":neg_table_id}
+        neg_tables.append(pos_table_id)
+
+    json.dump(question_table_id_dict,open("data/ottqa/triples_dict_"+split+".json","w"))
+    
```

Comment:
Based on these. hardwired paths,  there is no option to create the data directory anywhere but in the current directory

Reply:
Unfortunately yes, it has dependencies on a lot of other data like wiki links, indexes etc.